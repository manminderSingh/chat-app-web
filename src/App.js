import React, { useState, useEffect } from 'react';
import { Switch, Route, Link } from 'react-router-dom';

import AuthService from './../src/services/auth.service';

import Login from './components/auth/Login';
import Register from './components/auth/Register';
import Dashboard from './containers/Dashboard';

import "bootstrap/dist/css/bootstrap.min.css";
import './App.scss';

function App() {
  const [currentUser, setCurrentUser] = useState(undefined);

  useEffect(() => {
    const user = AuthService.getCurrentUser();
    if (user) {
      setCurrentUser(user);
    }
  }, []);

  const logout = () => {
    AuthService.logout();
  }

  return (
    <div className="App">
      <nav className="navbar navbar-expand navbar-dark bg-dark">
        <Link to={"/dashboard"} className="navbar-brand">
          ChatApp
        </Link>
        {currentUser ? (
          <div className='navbar-nav ml-auto'>
            <li className='nav-item'>
              <Link to={"/dashboard"} className="nav-link">
                {currentUser.username}
              </Link>
            </li>
            <li className="nav-item">
              <a href="/login" className="nav-link" onClick={logout}>
                LogOut
              </a>
            </li>
          </div>
        ) : (
          <div className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to={"/login"} className="nav-link">
                Login
              </Link>
            </li>

            <li className="nav-item">
              <Link to={"/register"} className="nav-link">
                Sign Up
              </Link>
            </li>
          </div>
        )}
      </nav>
      
      <div className="container mt-3">
          <Switch>
            <Route exact path={["/"]} component={Login} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/dashboard" component={Dashboard}/>
          </Switch>
      </div>
    </div>
  );
}

export default App;
