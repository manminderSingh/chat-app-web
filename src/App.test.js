import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import App from './App';
import './index.scss';

test('renders to match dom tree', () => {
  const component = renderer.create((<BrowserRouter><App /></BrowserRouter>));
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
