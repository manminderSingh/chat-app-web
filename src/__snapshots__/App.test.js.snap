// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match dom tree 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/dashboard"
      onClick={[Function]}
    >
      ChatApp
    </a>
    <div
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/register"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </li>
    </div>
  </nav>
  <div
    className="container mt-3"
  >
    <div
      className="col-md-12"
    >
      <div
        className="card card-container"
      >
        <img
          alt="profile-img"
          className="profile-img-card"
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
        />
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <div>
              <input
                className="form-control"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <div>
              <input
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn btn-primary btn-block"
              disabled={false}
            >
              <span>
                Login
              </span>
            </button>
          </div>
          
          <button
            disabled={true}
            style={
              Object {
                "display": "none",
              }
            }
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
